### ISSUE TITLE
NOZErrorCodeUnzipCannotDecompressFileEntry when decompressing a zip file

### ISSUE THREAD

**Issue by groue** [NSProgrammer/ZipUtilities#16]
Hello,

On iOS I get the `NOZErrorCodeUnzipCannotDecompressFileEntry` (18) error when I try to unzip a regular file zipped on a mac with the regular Finder command "Compress elements".

Precisely speaking, it looks like you get a Z_BUF_ERROR at the following line:
https://github.com/NSProgrammer/ZipUtilities/blob/1.6.6/ZipUtilities/NOZDeflateCoders.m#L321

I run the version 1.6.6 plus the patch from https://github.com/NSProgrammer/ZipUtilities/commit/c875b0413201f8c2e2b6ce6adbfa0fc3669a7be2

I decompress like that:

``` objc
NOZDecompressRequest *request = [[NOZDecompressRequest alloc] initWithSourceFilePath:url.path destinationDirectoryPath:decompressionURL.path];
NOZDecompressOperation *op = [[NOZDecompressOperation alloc] initWithRequest:request completion:^(NOZDecompressOperation * _Nonnull op, NOZDecompressResult * _Nonnull result) {
    if (!result.didSucceed) {
        // Get NOZErrorCodeUnzipCannotDecompressFileEntry (18)
    }
}];
```

**Comment by groue** [comment]
This may be a relevant link: http://www.zlib.net/zlib_faq.html#faq05

**Comment by NSProgrammer** [comment]
I have unit tests built around zip files that are created with "Compress elements", but I'm guessing these are not triggering the issue you are seeing.  Can you provide me a zip file that reproduces this?  Does it repro on iOS simulator, iOS device, and/or Mac OS X?  What OS version is it reproducing on?  The more details the better so I can target fixing something that repros and then bump the version to 1.7

Thanks for filing the issue and for the FAQ link.  I hope I can resolve this in short order once you send me a zip file that reproduces the issue.

**Comment by NSProgrammer** [comment]
I found a way to repro... basically needed a really large file compressed.  Looking at the issue now, thanks

**Comment by NSProgrammer** [comment]
Fixed in PR #17

**Comment by groue** [comment]
Do you still need a sample file ?

**Comment by NSProgrammer** [comment]
If you can repro with my latest change, yes.  But I'm pretty sure I fixed it with 1.7.0
